provider/image.go

```diff
@@ -212,7 +218,12 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		}
 
 		// we use the credentials for the server declared in the program, looking them up from the host authConfigs.
-		pushAuthConfig = authConfigs[registryServer]
+		cliPushAuthConfig, err := cfg.GetAuthConfig(registryServer)
+		if err != nil {
+			return "", nil, err
+		}
+
+		pushAuthConfig = types.AuthConfig(cliPushAuthConfig)
```

Comment:
The concrete types that implement the interface providing `GetAuthConfig` perform the `https://` fallback lookup, so we trust the library's behavior here.